cmd/fs-v1.go

```diff
@@ -113,12 +113,55 @@ func fsReaddirMetaBuckets(fsPath string) ([]string, error) {
 	return f.Readdirnames(-1)
 }
 
+// List of all bucket metadata configs.
 var bucketMetadataConfigs = []string{
 	bucketNotificationConfig,
 	bucketListenerConfig,
 	bucketPolicyConfig,
 }
 
+func migrateBucketMetadataConfigs(metaBucket, bucket, tmpBucket string) error {
+	for _, bucketMetaFile := range bucketMetadataConfigs {
+		fi, err := fsStatDir(pathJoin(metaBucket, tmpBucket, bucketMetaFile))
+		if err != nil {
+			switch errorCause(err) {
+			case errVolumeAccessDenied:
+			case errVolumeNotFound:
+			default:
+				return err
+			}
+		}
+
+		// There are no such files or directories found, proceed
+		// to next bucket metadata config.
+		if errorCause(err) == errVolumeNotFound {
+			continue
+		}
+
+		// Bucket metadata is a file.
+		if errorCause(err) == errVolumeAccessDenied {
+			if err = fsRenameFile(pathJoin(metaBucket, tmpBucket, bucketMetaFile),
+				pathJoin(metaBucket, bucket, bucketMetaFile)); err != nil {
+				if errorCause(err) != errFileNotFound {
+					return err
+				}
+			}
+			continue
+		} // else { bucket metadata is a directory.
+		if fi.IsDir() {
+			if err = fsRenameFile(pathJoin(metaBucket, tmpBucket, bucketMetaFile),
+				pathJoin(metaBucket, bucket, objectMetaPrefix, bucketMetaFile)); err != nil {
+				if errorCause(err) != errFileNotFound {
+					return err
+				}
+			}
+		}
+	}
+
+	// Success.
+	return nil
+}
```

Comment:
The problem with this that it is not catering for the issues we address with fs-v1-helpers.go readability doesn't change when we use a mix of os.Stat() and fsRenameFile() - it creates more confusion.  Going to add a lower level variant of `fsStat()` to be used by both fsStatFile and fsStatDir.